pkgs/development/python-modules/catboost/default.nix

```
@@ -0,0 +1,51 @@
+{ lib, buildPythonPackage, clang_7, fetchFromGitHub, graphviz, matplotlib, numpy
+, pandas, plotly, python2, python3, pythonOlder, scipy, six, withCuda ? false
+, cudatoolkit }:
+
+buildPythonPackage {
+  pname = "catboost";
+  version = "0.24.4";
+
+  disabled = pythonOlder "3.4";
+
+  src = fetchFromGitHub {
+    owner = "catboost";
+    repo = "catboost";
+    rev = "2cb3eda7dd228e851d9817eb018b9d7521ac342d";
+    sha256 = "sha256-pzmwEiKziB4ldnKgeCsP2HdnisX8sOkLssAzNfcSEx8=";
+  };
+
+  # Passed to make in setup.py.
+  nativeBuildInputs = [ clang_7 python2 ];
+
+  propagatedBuildInputs = [ graphviz matplotlib numpy pandas scipy plotly six ]
+    ++ lib.optional withCuda [ cudatoolkit ];
+
+  # First apply patches, then go to source root; sourceRoot applies too early.
+  patches = [ ./nix-support.patch ./format.patch ];
+  postPatch = "cd catboost/python-package";
+
+  setupPyBuildFlags = [ "--with-ymake=no" ];
+  CUDA_ROOT = lib.optional withCuda cudatoolkit;
+
+  # Tests use custom "ya" tool, not yet supported.
+  dontUseSetuptoolsCheck = true;
+
+  # Would use setup.py --parallel=..., but setupPyBuildFlags are quoted.
+  MAKEFLAGS = "-j$(NIX_BUILD_CORES)";
```

```suggestion
  MAKEFLAGS = "-j$NIX_BUILD_CORES";
```

I think caused some issue why my machine froze.

---

Your suggestion effetively disables parallel build. Used preBuild to set MAKEFLAGS. On my 8-core machine it speeds up the build without any freezes.

---

NIX_BUILD_CORE is a global variable by nix and not a command. Maybe the space is not optional like I thought but the () are definitely to much here. Maybe your nix-build-cores settings is wrong.

---

Modified nix support patch, so that setup.py picks up NIX_BUILD_CORES and enableParallelBuilding just works.